lib/ClangImporter/ImporterImpl.h

```diff
@@ -418,6 +418,9 @@ class LLVM_LIBRARY_VISIBILITY ClangImporter::Implementation
   /// Mapping of already-imported declarations.
   llvm::DenseMap<std::pair<const clang::Decl *, Version>, Decl *> ImportedDecls;
 
+  /// Set of failed-imported declarations.
+  llvm::DenseSet<std::pair<const clang::Decl *, Version>> FailedImportedDecls;
```

Comment:
So, I think the main question here is whether we want to have both `FailedImportedDecls` and `ImportedDecls` or just put everything in `ImportedDecls`. I think it might be a bit simpler just to have one place where everything cached (especially because it is sometimes checked in the individual visitors).

Reply:
Sure, I wasn't sure if we wanted to modify ImportedDecls or just importedDeclCached so I opted for importDeclCached initially because it appeared more local (used in ~4 places while ImportedDecls in ~30). 

However I suppose we can store a nullptr in the map, and then importDeclCached would return None if not found and a nullptr if  found and failed.